src/typescriptGenerator.ts

```diff
@@ -67,10 +67,13 @@ export async function generateTypeScriptLibrary(
     version: (await host.GetValue("package-version")) || "1.0.0"
   };
 
+  const shouldGenerateLicenseHeader: boolean =
+    (await host.GetValue("license-header")) || false;
+
   // Skip metadata generation if `generate-metadata` is explicitly false
   if ((await host.GetValue("generate-metadata")) !== false) {
     generatePackageJson(clientDetails, packageDetails, project);
-    generateLicenseFile(project);
+    if (shouldGenerateLicenseHeader) generateLicenseFile(project);

```

Comment:
nit: you could pass `shouldGenerateLicenseHeader` to `generateLicenseFile` and check it there, just to maintain the format of just calling the generate functions

Reply:
Done